download-and-preprocessing.kts

```diff
@@ -15,23 +21,26 @@ rename("./temp/phosphor-icons/SVGs Flat/light", "light")
 rename("./temp/phosphor-icons/SVGs Flat/regular", "")
 rename("./temp/phosphor-icons/SVGs Flat/thin", "thin")
 
-fun download() {
-    val phosphorIconDownloadUrl = "https://phosphoricons.com/assets/phosphor-icons.zip"
-
-
-    if (!workDir.exists()) {
-        println("create /temp folder")
-        workDir.mkdirs()
+for (file in File("./temp/phosphor-icons/SVGs Flat").walk()) {
+    if (file.isFile) {
+        setSvgSizeTo24(file)
     }
-    val saveFile = File(workDir, "phosphor-icons.zip")
+}
 
-    val url = URL(phosphorIconDownloadUrl)
+fun downloadFile(targetFile: File, downloadUrl: String) {
+    val parentDir = targetFile.parentFile
+    if (parentDir.exists()) {
+        deleteDirectory(parentDir)
+    }
+    println("creating ${parentDir}")
+    parentDir.mkdirs()
+    val url = URL(downloadUrl)
     val connection = url.openConnection() as HttpURLConnection
```

Comment:
Could we import and use Ktor here instead (just for code clarity), or is it too overkill 😅

Reply:
It should be okay. I can try it.